### PR TITLE
VIM-4027: vimaccountnew → vimaccount

### PR DESCRIPTION
**Pull request by huebnerob**
# Ticket(s)

https://vimean.atlassian.net/browse/VIM-4027

# Ticket Summary

We currently have two model objects that represent accounts: an old VIMAccount and the modern VIMAccountNew.  This is due to a change made in version 5.5 of the app (approximately June 2015 vintage) where we wanted to seamlessly and invisibly migrate user's accounts over so they didn't need to log in again.

Well, app versions lower than v5.5 account for less than half of a percent of our daily sessions now (under 1000 users), so I think it's safe to move past this compatibility bridge, especially as we're now rethinking our networking library.

On the off chance that any of these legacy users actually migrate to a newer version in the future, account retrieval will fail silently and the user will have to log in again.

# Implementation Summary

* Removed VIMAccount (old) and some associated legacy methods
* Renamed VIMAccountNew to VIMAccount

# How to test

Login, logout

### CODE REVIEW

**Comment by alfiehanssen** [comment]
@huebnerob looks like we'll release this into the wild well before VimeoNetworking, nice

**Comment by alfiehanssen** [comment]
@huebnerob I don't think we can delete VIMAccountNew, we'll run into the same problem. 

VIMAccountNew is what's archived in every user out there, which means that if we delete that class we won't be able to unarchive those accounts, pretty sure everyone will be logged out, right?

**Comment by huebnerob** [comment]
@alfiehanssen Good catch! I overlooked the sensitivity of NSCoding to class names.  I've added a fix that sets the newly renamed `VIMAccount` class as the target for unarchiving "VIMAccountNew" objects.

@nicolelehrer ready for review!

**Comment by nicolelehrer** [comment]
@huebnerob DWR 👍

File: VIMNetworking/Model/VIMAccount.m [hunk 1]
```diff
@@ -42,45 +71,38 @@ + (BOOL)supportsSecureCoding
...
+    [NSKeyedUnarchiver setClass:self forClassName:@"VIMAccountNew"];
+}
+
 - (id)initWithCoder:(NSCoder *)aDecoder
 {
     self = [super init];
     if(self)
     {
-        // TODO: need to use decodeObjectOfClass but this is type id, what to do? [AH]
+        self.accessToken = [aDecoder decodeObjectOfClass:[NSString class] forKey:@"accessToken"];
```
Comment (alfiehanssen) [inline VIMNetworking/Model/VIMAccount.m]: @huebnerob Is there a separate PR that leverages: `NSStringFromSelector(@selector(accessToken))`?

**Pull request merged**